src/vm/jithelpers.cpp

```diff
@@ -5723,6 +5723,14 @@ Thread * __stdcall JIT_InitPInvokeFrame(InlinedCallFrame *pFrame, PTR_VOID StubS
 
 #endif // _WIN64
 
+EXTERN_C void JIT_PInvokeBegin(InlinedCallFrame* pFrame);
+EXTERN_C void JIT_PInvokeEnd(InlinedCallFrame* pFrame);
+
+EXTERN_C void STDCALL RareDisablePreemptiveGCHelper(Thread* pThread)
```

Comment:
Ideally, this would do the same thing as `JIT_RareDisableHelper`. Can we just call `JIT_RareDisableHelper` instead of this?

E.g. `JIT_RareDisableHelper` has `BEGIN_PRESERVE_LAST_ERROR` / `END_PRESERVE_LAST_ERROR` that I think will be needed here as well.

Reply:
I'm not too sure. I can use the existing helper and test how this would work.

Reply:
Replaced with the existing helper